k8sdeps/transformer/target/target.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package target
+
+import (
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/kustomize/v3/pkg/gvk"
+	"sigs.k8s.io/kustomize/v3/pkg/resource"
+	"sigs.k8s.io/kustomize/v3/pkg/types"
+)
+
+// FindPatchTargets returns a list of resources that
+// are pointed by targets
+func FindPatchTargets(targets types.Targets,
+	rs []*resource.Resource) ([]*resource.Resource, error) {
+	rs = findPatchTargetsByNameSpace(targets.Namespace,
+		findPatchTargetsByName(targets.Name,
+			findPatchTargetsByGvk(targets.Gvk, rs)))
+
+	rs, err := findPatchTargetsByLabels(targets.LabelSelector, rs)
+	if err != nil {
+		return nil, err
+	}
+	rs, err = findPatchTargetsByAnnotations(targets.AnnotationSelector, rs)
+	if err != nil {
+		return nil, err
+	}
+	return rs, nil
+}
+
+// findPatchTargetsByLabels returns a list of resources that
+// match with a label selector
+func findPatchTargetsByLabels(selector string,
+	rs []*resource.Resource) ([]*resource.Resource, error) {
+	s, err := labels.Parse(selector)
```

Comment:
please bury the `labels` package dependence inside `UnstructAdapter`.

Add a method to `Kunstructured`, e.g.
```
type Kunstructured interface {
	MatchesLabelSelector(selector string) bool
}
```
(or something)

Reply:
Added two function
```
MatchesLabelSelector(selector string) (bool, error)
MatchesAnnotationSelector(selector string) (bool, error)
```